packages/delisp-core/src/unify.ts

```diff
@@ -205,7 +214,11 @@ export function unify(
   ctx: Substitution = {}
 ): UnifyResult {
   // RULE (uni-const)
-  if (t1.type === "string" && t2.type === "string") {
+  if (t1 === undefined) {
```

Comment:
I would not include this check as part of `unify`. It also violates the type specification of `unify`. Can we instead identify which function is calling `unify` wrongly instead (`unifyArray` I pressume)?

Reply:
I started adding this check to `unifyArray` at first but was thinking that putting it in `unify` might solve the issue for more cases. But you have a good point and it’s indeed a lot nicer to put the check at the caller side.